tests/puzzles/test_puzzles.py

```diff
@@ -0,0 +1,234 @@
+import hashlib
+
+from typing import Iterable, List, Tuple
+from unittest import TestCase
+
+from blspy import AugSchemeMPL, BasicSchemeMPL, G1Element, G2Element
+
+from src.types.blockchain_format.program import Program
+from src.types.blockchain_format.sized_bytes import bytes32
+from src.types.coin_solution import CoinSolution
+from src.types.spend_bundle import SpendBundle
+from src.util.condition_tools import ConditionOpcode
+from src.wallet.puzzles import (
+    p2_conditions,
+    p2_delegated_conditions,
+    p2_delegated_puzzle,
+    p2_puzzle_hash,
+    p2_m_of_n_delegate_direct,
+    p2_delegated_puzzle_or_hidden_puzzle,
+)
+from tests.util.key_tool import KeyTool
+
+from .coin_store import CoinStore, CoinTimestamp
+
+
+DEFAULT_KEYCHAIN = KeyTool()
+
+HIERARCHICAL_PRIVATE_KEY = BasicSchemeMPL.key_gen(hashlib.sha256(b"foo").digest())
+
+GROUP_ORDER = 0x73EDA753299D7D483339D80809A1D80553BDA402FFFE5BFEFFFFFFFF00000001
+
+
+T1 = CoinTimestamp(1, 10000000)
+T2 = CoinTimestamp(5, 10003000)
+
+
+def secret_exponent_for_index(index: int) -> G2Element:
```

Comment:
This is not necessary, our bls library has the PrivateKey class.
You can do sk.get_g2()

Reply:
Also we use std_hash method

Reply:
The signature is a lie (not sure why it wasn't caught by `mypy`)... if you look, you'll see it actually returns an `int`.

Reply:
This isn't designed to be compatible with anything, just a quick hack to generate a throwaway secret exponent. Truthfully, none of this code actually even checks signatures, just conditions, so I could probably remove all this code entirely.

Reply:
These files have moved here https://github.com/Chia-Network/chia-blockchain/pull/1083/files#diff-893cbcf9140f6c65a62c0239c529d94c68a482078cc00131e6272e12afa385b2